project-fortis-interfaces/src/components/Admin/UserRoles.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import Snackbar from 'material-ui/Snackbar';
+import groupBy from 'lodash/groupBy';
+import differenceBy from 'lodash/differenceBy';
+import { DataGrid } from './DataGrid';
+import { getColumns } from './shared';
+const { Editors, Formatters } = require('react-data-grid-addons');
+const { DropDownEditor } = Editors;
+const { DropDownFormatter } = Formatters;
+
+const roles = ['user', 'admin'];
+
+class UserRoles extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: false,
+      message: ''
+    };
+
+    this.getUserRolesColumns = this.getUserRolesColumns.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleRemove = this.handleRemove.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.flux.actions.ADMIN.load_users()
+  }
+
+  getUserRolesColumns() {
+    const columnValues = [
+      {editable: true, filterable: true, sortable: true, key: "identifier", name: "Identity"},
+      {editable: false, filterable: true, sortable: true, key: "role", name: "Role", editor: <DropDownEditor options={roles}/>, formatter: <DropDownFormatter options={roles}/>}
+    ];
+
+    return getColumns(columnValues);
+  }
+
+  handleSave(users) {
+    const adminRowsMissingUserRole = this.getAllAdminWithoutUserRole(users);
+    if (adminRowsMissingUserRole.length > 0) this.alertAddUserOnAddAdmin(adminRowsMissingUserRole);
+    this.props.flux.actions.ADMIN.add_users(this.removeIdsFromUsers(users));
+  }
+
+  getAllAdminWithoutUserRole(users) {
+    const usersGroupedByRole = groupBy(users, 'role');
+    return differenceBy(usersGroupedByRole.admin, usersGroupedByRole.user, 'identifier');
+  }
+
+  alertAddUserOnAddAdmin(admin) {
+    const adminIdentities = admin.map(a => a.identifier);
```

Comment:
NB: underscore has a [pluck](http://underscorejs.org/#pluck) method that does just this map to get properties. Not saying we need to use it, just wanted to point it out since there are a few other nice underscore uses in this module.